### DISCUSSION TITLE
Is my GPS working?

### DISCUSSION THREAD

**Discussion by ldb20151** [aaronwmorris/indi-allsky#1634] [Q&A] [answered]
All, I have Indi AllSky up and running locally, which is a huge credit to the work Aaron and the community put into it.  I'm no RPI expert and don't speak linux/raspian.  Thanks everyone!  That said, I know I have work ahead to get it tuned and the functions set up the way I want.  Hopefully that will continue to go ok.  

At this point, all seems to be working OK, but like I said I'm at the beginning of the activation process.  I can visually see the camera is working pretty much as intended; however I'm wondering if I should trust the GPS coordinates displayed on the screen.  I installed a GPS hat on my RPI 5.  As I understand, Indi Allsky includes the files necessary to activate and use the GPS, but I don't see any tools to specifically check it out and make sure its doing what it should.  The displayed values seems about right but could they just be derived from my IP or some other bit of data entered at setup?  

Am I correct that Indi Allsky installed GPSd and the GPS is active?  How can I test to be certain its working correctly?

v/r,
-Larry

**Comment by aaronwmorris** [comment, marked as answer]
indi-allsky does not install or setup GPSd by default.  I have a separate script to install GPSd.

```
./misc/setup_gpsd.sh
```

Once gpsd is working, you can verify it having a lock on satellites with `gpsmon`

There is also a wiki page to help with time syncing.  https://github.com/aaronwmorris/indi-allsky/wiki/GPSd-NTP-Time-Sync

**Reply by ldb20151** [reply]
I'm glad I asked -- not sure what I read that led me to believe the Indi AllSky scripts did it all.  I appreciate the info.  I'm really trying to sort this out myself, but not knowing what I don't know, I find its better to be cautious versus experimenting.  I don't yet know enough about the software side to fix issues without wiping the Pi and starting over.  Which isn't fun!

**Reply by aaronwmorris** [reply]
It is no big deal.  indi-allsky is a huge project which requires specialization across multiple disciplines.

**Comment by Aripaule** [comment]
hello, 

I followed the steps on this page,

https://github.com/aaronwmorris/indi-allsky/wiki/GPSd-NTP-Time-Sync


however when you test the cgps command, no data is received.
![Imagem do WhatsApp de 2024-11-23 à(s) 21 08 49_eeb17d27](https://github.com/user-attachments/assets/8b591f2b-5b0b-46c0-b650-f5636eab0f66)


any idea what it could be?

**Reply by aaronwmorris** [reply]
What GPS hat are you using?

**Reply by Aripaule** [reply]
![image](https://github.com/user-attachments/assets/a447d7f4-de20-44fa-b919-130a7635967a)

**Reply by aaronwmorris** [reply]
You will need to run `raspi-config` to activate the TTL serial port on pins GPIO 14 (tx) & 15 (rx) (pins 8 & 10 respectively).  When you wire the serial pins, make sure you wire TX to RX and RX to TX.

You then will need to edit `/etc/default/gpsd` to say `DEVICES=/dev/serial0` I believe (although it might be /dev/ttyS0).  Then restart gpsd with `sudo systemctl restart gpsd`.

**Reply by Aripaule** [reply]
![Imagem do WhatsApp de 2024-11-24 à(s) 08 22 49_043e131e](https://github.com/user-attachments/assets/323e6291-7273-4898-8db5-432c237f48e3)


I tried both and the error remains the same.

DEVICES=/dev/serial0

DEVICES=/dev/ttyS0

I also activated a serial port in raspi-config.

and I double checked wiring (tx- rx/ rx-tx)

**Reply by aaronwmorris** [reply]
Check the output of `sudo dmesg` to see if it mentions the names of the serial ports.

**Reply by Aripaule** [reply]
> Check the output of `sudo dmesg` to see if it mentions the names of the serial ports.

hi, 
the command generated a huge checklist.

[log.txt](https://github.com/user-attachments/files/17963480/log.txt)

**Reply by aaronwmorris** [reply]
I think you have configured `/dev/ttys0` but I would try `/dev/ttyS0`  (capital S)

**Reply by Aripaule** [reply]
I corrected the "S" and now it is pinging for 400 seconds. Did it work?
![Imagem do WhatsApp de 2024-11-29 à(s) 21 14 27_b3b0f762](https://github.com/user-attachments/assets/0a2645cf-6773-4ac9-8f81-c15eed8d1723)

**Reply by aaronwmorris** [reply]
It is looking better.  You may want to walk the system outside to see if maybe being inside is preventing a full connection.

**Reply by Aripaule** [reply]
Hi Aaron, I hope I'm not bothering you with so many questions.


But I still haven't gotten a connection. Now the module is in my backyard.
![Imagem do WhatsApp de 2024-12-01 à(s) 18 39 57_60258440](https://github.com/user-attachments/assets/867e39ac-7cdf-4f33-bdd7-7b7d02a0e30d)

**Comment by ldb20151** [comment]
Thank you. Great timing as I’m working on this today. 🤞LarrySent from Larry's iPhoneOn Nov 24, 2024, at 11:33 AM, Aaron W Morris ***@***.***> wrote:﻿
Check the output of sudo dmesg to see if it mentions the names of the serial ports.

**Reply by aaronwmorris** [reply]
After letting it run for a while, can you post the output of `sudo dmesg` again?

**Reply by Aripaule** [reply]
[log. gps.txt](https://github.com/user-attachments/files/17970751/log.gps.txt)

**Reply by aaronwmorris** [reply]
Okay, can you show the output of `id gpsd` ?

**Reply by Aripaule** [reply]
ari@ari-desktop:~$ id gpsd
uid=123(gpsd) gid=20(dialout) groups=20(dialout)

**Reply by aaronwmorris** [reply]
Okay, the gps module is supposed to run at 9600 baud.  Lets try setting the speed.

Edit `/etc/default/gpsd` and set the GPSD_OPTIONS to the following (keep the other settings)

```
GPSD_OPTIONS="-n -s 9600"
```

**Reply by Aripaule** [reply]
ok, it doesn't seem to be any changes with command

-cgps

**Reply by aaronwmorris** [reply]
You will need to make sure to restart gpsd if you made this change.

```
sudo systemctl restart gpsd
```

If that does not fix it... you may want to try reversing the RX/TX lines to see if maybe that is the problem.

**Reply by aaronwmorris** [reply]
I was reading a page about setting up a GPS adapter using the raspberry PI serial port and it seems the correct device name might be `/dev/serial0`.  See if that helps with your config.

**Reply by Aripaule** [reply]
Hi, I tried doing what you said, but it doesn't seem to have had any effect.

It is freezed:
![image](https://github.com/user-attachments/assets/44c52620-f57c-4089-bfd5-7736fdba325e)

log with sudo dmesg 

[gps 2.txt](https://github.com/user-attachments/files/18030978/gps.2.txt)

**Reply by aaronwmorris** [reply]
Okay, there is one other serial port in your log you can try.

`/dev/ttyAMA1`

**Reply by Aripaule** [reply]
same as before

**Reply by aaronwmorris** [reply]
Did you remove the baud rate that I had you try previously?

**Reply by Aripaule** [reply]
no, i dont. let me check it. brb asap

**Reply by Aripaule** [reply]
![image](https://github.com/user-attachments/assets/b970fe0d-f6e5-4712-9972-1d4c5a032b01)


> Did you remove the baud rate that I had you try previously?


only with the ttyS0 serial port did the prompt show any changes. Even removing the baud rate in the other two as you suggested.

![image](https://github.com/user-attachments/assets/cd11cf8d-e7d9-43b6-9d58-1bbd3ef0f243)

**Reply by aaronwmorris** [reply]
The output from `ttyS0` is more of what I would expect.  That is what it looks like when you cannot get a GPS lock.  If you are testing this inside, you are probably getting a bad signal.

Try moving the antenna around.  It can take a few minutes for the GPS to get a good lock.

**Reply by Aripaule** [reply]
I tested it outside, but I didn't get any improvement, (but I had the baud rate activated).

I will test again as soon as weather permits. Do you think I should keep this serial port or switch to the others?

**Reply by aaronwmorris** [reply]
I think what you are experiencing is probably kind of normal.  Most of these GPS devices are all the same, just sold by different manufacturers.

**Reply by ldb20151** [reply]
I'm probably not helping but for what it's worth, my new GPS hat serial port was "ttyAMA0" (last character is a zero) which wasn't consistent with their install manual.  Not sure what establishes this.  
Good Luck!

**Reply by Aripaule** [reply]
> I'm probably not helping but for what it's worth, my new GPS hat serial port was "ttyAMA0" (last character is a zero) which wasn't consistent with their install manual. Not sure what establishes this. Good Luck!

I didn't get any improvement in the problem. I bought another  GPS module to ensure that my unit is not defective. As soon as I can test it I'll come back here.

**Comment by Aripaule** [comment]
ok, no improvement even leaving the antenna exposed, and even far from the pi (I heard that it could be some interference from the Wi-Fi)

The last solution is to buy another sensor, as soon as it arrives I will let you know.

**Reply by JimTaylor-Iowa** [reply]
what sensor did you ultimately buy for the replacement...

**Reply by Aripaule** [reply]
hi,

none of them. I gave up GPS
